rclcpp/include/rclcpp/timer.hpp

```diff
@@ -149,11 +149,40 @@ class TimerBase
   bool
   exchange_in_use_by_wait_set_state(bool in_use_state);
 
+  /// Set a callback to be called when the timer is reset
+  /**
+   * You should aim to make this callback fast and not blocking.
+   * If you need to do a lot of work or wait for some other event, you should
+   * spin it off to another thread.
+   *
+   * Calling it again will override any previously set callback.
+   *
+   * This function is thread-safe.
```

Comment:
```suggestion
   * An exception will be thrown if the callback is not callable.
   *
   * This function is thread-safe.
   *
   * If the timer has already been reset, the callback will be called.
```

Reply:
Done in [4d26bcc](https://github.com/ros2/rclcpp/pull/1979/commits/4d26bccfc2fdffe77a8fd04de630bdc148b27624)